### DISCUSSION TITLE
Does the DTO generated by Add ServiceStack Refrences support namespaces?

### DISCUSSION THREAD

**Discussion by mylvghb** [ServiceStack/Discuss#87] [Q&A] [answered]
The DTOs in the ServiceModel project may be stored in different namespaces according to the database tables (the same tables may exist in different schemas), so when generating the DTOs through Add Service Refrences, will they be stored in different namespaces according to the DTO namespace on our Server?

Sometimes, due to larger projects and more DTOs, namespaces are used to separate the business.

Server DTOs:

```csharp
namespance A{
   public class GetProduct
   {
       public int Id{get;set;}
   }
}
namespance B{
   public class GetProduct
   {
       public int Id{get;set;}
   }
}
```
If these DTOs are stored in a single file on the Client and not in a different namespace/folder it will cause a conflict with the same name.

**Comment by mythz** [comment, marked as answer]
Only .NET languages support namespaces (when not using a GlobalNamespace), in [all other languages DTOs need to be unique](https://docs.servicestack.net/add-servicestack-reference#limitations).
